### PR TITLE
Feature/28_amenities - only-owner restriction

### PR DESCRIPTION
**Pull request by zilola08**
- added function "verifyOwner" to check if the user is the owner of the accommodation and added "only-owner" restriction to the add/update/delete amenities functions;
- as discussed with the team, removed name from addBearerAuth() in the amenities part;
- deleted console.logs;

### CODE REVIEW

File: src/amenities/amenities.controller.ts [hunk 1]
```diff
@@ -91,12 +91,12 @@ export class AmenitiesController {
     description: 'Internal Server Error',
   })
   @ApiBody({ type: AmenitiesDto })
-  addAmenities(@Param('id') id: string, @Body() dto: AmenitiesDto) {
-    return this.amenitiesService.addAmenities(id, dto);
+  addAmenities(@Param('id') id: string, @Body() dto: AmenitiesDto, @Req() req: any) {
+    return this.amenitiesService.addAmenities(id, dto, req.user.id);
   }
```
Comment (mirsadikov) [inline src/amenities/amenities.controller.ts:96]: There is a CurrentUser decorator to get user id:

`@CurrentUser('id') userId: string`

**Review by mirsadikov — CHANGES_REQUESTED** [review]

File: src/amenities/amenities.controller.ts (hunk 1 above)
Comment (zilola08) [reply src/amenities/amenities.controller.ts:101]: thanks, changed it

File: src/amenities/amenities.service.ts [hunk 2]
```diff
@@ -10,6 +15,24 @@ import PrismaErrorCodes from 'src/errors/prismaErrorCodes.enum';
...
+        where: {
+          id: accommodationId,
+        },
+        select: {
+          ownerId: true,
+        },
+      });
+      if (!accommodation) throw new NotFoundException('No accomodation found with this id');
+      return userId === accommodation?.ownerId;
+    } catch (error) {
```
Comment (Zach13131) [inline src/amenities/amenities.service.ts:30]:
```
 try {
      const accommodation = await this.prisma.accommodation.findUnique({
        where: {
          id: accommodationId,
          ownerId: userId,
        },
      });
      return accommodation;
    } catch (error) {
      throw new GlobalException(ErrorsTypes.YOUR_ERROR);
    }
```
it is much better, to check exsisting accommodation and owner at once. and remove check for instance in catch block

File: src/amenities/amenities.service.ts [hunk 3]
```diff
@@ -46,7 +69,9 @@ export class AmenitiesService {
     }
   }
 
-  async addAmenities(id: string, dto: AmenitiesDto) {
+  async addAmenities(id: string, dto: AmenitiesDto, userId: string) {
+    const isOwner = await this.verifyOwner(userId, id);
+    if (!isOwner) throw new UnauthorizedException('You are not authorized to perform this action');
```
Comment (Zach13131) [inline src/amenities/amenities.service.ts:74]: And now throw NotFoundException('Accomodation with this id and ownerId are not found'), with the message like this.

**Review by Zach13131 — CHANGES_REQUESTED** [review]

**Comment by zilola08** [comment]
moved owner restriction into prisma query

**Review by mirsadikov — APPROVED** [review]
Looks good for me now

**Review by Zach13131 — APPROVED** [review]
it is even better.

**Review by gitkoshelew — APPROVED** [review]

**Pull request merged**